chemprop/train/molecule_fingerprint.py

```diff
@@ -76,7 +76,12 @@ def molecule_fingerprint(args: FingerprintArgs, smiles: List[List[str]] = None)
 
     # Set fingerprint size
     if args.fingerprint_type == 'MPN':
-        total_fp_size = args.hidden_size * args.number_of_molecules
+        if args.atom_descriptors == "descriptor": # special case when we have 'descriptor' extra dimensions need to be added
+            total_fp_size = (args.hidden_size + test_data.atom_descriptors_size()) * args.number_of_molecules
+        else:
+                total_fp_size = args.hidden_size * args.number_of_molecules
+        if args.atom_descriptors == "feature":
```

Comment:
There is another bit that could be added to make it compatible with extra bond features too. Could you remove your new lines 83 and 84 and then add these lines from `make_predictions/set_features` between lines 38 and 39?

```
    if args.atom_descriptors == 'feature':
        set_extra_atom_fdim(train_args.atom_features_size)

    if args.bond_features_path is not None:
        set_extra_bond_fdim(train_args.bond_features_size)
```